Lesson9/main.js

```diff
@@ -0,0 +1,47 @@
+const ul = document.getElementById("js-ul");
+const loading = document.getElementById("js-loading")
+const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}];
```

Comment:
Q
Do these variables need to be global?

Reply:
Thank you for your comment Mai-san,

It did not have to be variable for my original code.
Yet, after refactoring some code, promise function also needs to retrieve the value of features array, so I left it as it was 
Please check my latest commit 28cc47a6943db8e2233c1d0dbabc798be71deb5f

Reply:
I'm sorry, maybe I didn't ask the right questions. 

`const loading = document.getElementById("js-loading")`
`const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]; `

I think these variables can be local variables instead of global variables. 
Declaring them in a function can reduce the scope and make the code less prone to bugs.

I couldn't find a very good article about scope, but here are some articles
https://jsprimer.net/basic/function-scope/#reduce-scope

```JavaScript
const addLoading = () =>{
  const loading = document.getElementById("js-loading")
  const loadingImg = document.createElement('img');
  loadingImg.src = 'loading.gif';
  loadingImg.id = "loading-img";
  loading.appendChild(loadingImg);
};
```
```JavaScript
const receiveData = new Promise((resolve) =>{
  addLoading();
  const features = [
    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
  ]; 
  setTimeout(() => resolve(features), 3000)
});
```

Reply:
Thank you for sharing the article and showing me the example codes.
I thought features array had to be a global variable because there is createLists() in the asyncFeature function, and I though it would need to access to the features array to iterate through the data. But looking at the code again, the argument passed in is "result" that was declared and passed from the receiveData(), which already has the data of features array.
It became a lot more clear now. Thank you Mai-san,
Kindly check my latest commit